src/main/java/com/epam/aidial/core/util/Compression.java

```diff
@@ -29,27 +29,23 @@ public byte[] compress(String type, byte[] data) {
     }
 
     @SneakyThrows
-    public byte[] decompress(String type, InputStream input) {
+    public InputStream decompress(String type, InputStream input) {
         if (!type.equals("gzip")) {
             throw new IllegalArgumentException("Unsupported compression: " + type);
         }
 
-        try (GZIPInputStream stream = new GZIPInputStream(input)) {
-            return stream.readAllBytes();
-        }
+        return new GZIPInputStream(input);
```

Comment:
How do we handle this special case for GCP?

Reply:
Fixed.